lib/hexo/post.js

```diff
@@ -55,14 +55,14 @@ class PostRenderCache {
   }
 }
 
-const prepareFrontMatter = data => {
+const prepareFrontMatter = (data, jsonMode) => {
   for (const [key, item] of Object.entries(data)) {
     if (moment.isMoment(item)) {
       data[key] = item.utc().format('YYYY-MM-DD HH:mm:ss');
     } else if (moment.isDate(item)) {
       data[key] = moment.utc(item).format('YYYY-MM-DD HH:mm:ss');
     } else if (typeof item === 'string') {
-      data[key] = `"${item}"`;
+      if (jsonMode || item.includes(':') || item.startsWith('#') || item.startsWith('!!')) data[key] = `"${item}"`;
```

Comment:
Is it meets the YAML spec?

Reply:
Which operation do you mean? Currently (before this PR) string is always wrapped in double-quote, so yaml's comment and regex syntax in the front-matter are also always parsed as regular string. This PR retains the process.

Reply:
This line only affects when `item` is a string. if a scaffold's front-matter has a regex,

``` yml
---
title: {{ title }}
date: {{ date }}
tags:
foo: !!js/regexp /pattern/gim
typora: /{{ foo }}/
---
```

will be rendered as:

``` yml
---
title: bar
foo: !<tag:yaml.org,2002:js/regexp> /pattern/gmi
typora: //
date: 2020-09-10 02:46:20
tags:
---
```

In this example, `foo` is a regex, so L65 doesn't affect it.